### PR TITLE
fix: improve error messages context & format for PowerBI service account errors

### PR DESCRIPTION
**Pull request by csm-thu**

### CODE REVIEW

File: src/services/powerbi/errors.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  console.error('Unknown error:');
+  console.error(error);
+  return forgePowerBIServiceAccountError('', 'Unknown error', error.message);
+};
+
+export const handleUserAccountError = (error) => {
+  if (error instanceof PowerBIError)
+    return forgePowerBIUserAccountError(error.status, error.statusText, error.errorMessage, error.description);
+
+  if (error.response) {
```
Comment (nborde-CSM) [inline src/services/powerbi/errors.js:88]: Same as above

**Review by nborde-CSM — APPROVED** [review]
Some little remaks but LGTM :+1:

File: api/GetEmbedInfo/config.js [hunk 2]
```diff
@@ -1,18 +1,35 @@
...
-// system.loggerOptions.logLevel values
-// Error = 0,
-// Warning = 1,
-// Info = 2,
-// Verbose = 3,
-// Trace = 4;
+// Sanitize the provided value to always provided a non-empty string or undefined
+const sanitizeValue = (value) => {
+  const strValue = value?.trim() ?? '';
+  return strValue.length !== 0 ? strValue : undefined;
```
[the remark this answers is not in the document]
Comment (csm-thu) [reply api/GetEmbedInfo/config.js:7]: - `sanitizeValue` is only called on environment variables, that are strings or `undefined`
- as documented above the function we want "to always provide[...] a non-empty string or undefined" in this function, in order to fill the `envVars` dict later; returning `undefined` allows us to simplify the call below: `...ALL_PARAMETERS.map((parameterName) => ({ [parameterName]: sanitizeValue(process.env[parameterName]) }))`

File: src/services/powerbi/errors.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+export const forgePowerBINetworkError = (details) => {
+  const errorMessage =
+    "Can't reach the server to retrieve a service account token for powerBI. If the problem persists, please contact " +
+    'an administrator.';
+  return forgePowerBIServiceAccountError(504, 'Network error', errorMessage, details);
+};
+
+export const handleServiceAccountError = (error) => {
+  if (error.response) {
```
[the remark this answers is not in the document]
Comment (csm-thu) [reply src/services/powerbi/errors.js]: We only call `handleServiceAccountError` when we catch an error, or we build a custom error object, so `error` should never be `null` or `undefined` except for future devs mistakes :wink: 

Still, I added your suggestion to be future-proof :slightly_smiling_face:

File: src/services/powerbi/errors.js (hunk 1 above)
Comment (csm-thu) [reply src/services/powerbi/errors.js]: Fixed

**Pull request merged**